sktime/forecasting/base/_adapters.py

```diff
@@ -12,6 +12,72 @@
 from sktime.forecasting.base._sktime import _SktimeForecaster
 from sktime.forecasting.base._sktime import _OptionalForecastingHorizonMixin
 
+from sktime.utils.validation.forecasting import check_y
+
+
+class _TbatsAdapter(_OptionalForecastingHorizonMixin, _SktimeForecaster):
+    """Base class for interfacing tbats forecasting algorithms"""
+
+    def fit(self, y, X=None, fh=None):
+        """Fit to training data.
+
+        Parameters
+        ----------
+        y : pd.Series
+            Target time series to which to fit the forecaster.
+        fh : int, list or np.array, optional (default=None)
+            The forecasters horizon with the steps ahead to to predict.
+        X : pd.DataFrame, optional (default=None)
+            Exogenous variables (ignored)
+
+        Returns
+        -------
+        self : returns an instance of self.
+        """
+        if X is not None:
+            raise NotImplementedError("BATS/TBATS don't support exog or endog data.")
+
+        y = check_y(y)
+        self._set_y_X(y, X)
+        self._set_fh(fh)
+
+        self._forecaster = self._forecaster.fit(y)
+        self._is_fitted = True
+        return self
+
+    def _predict(self, fh, X=None, return_pred_int=False, alpha=DEFAULT_ALPHA):
+        if not fh.is_relative:
+            fh = fh.to_relative(cutoff=self.cutoff)
+
+        if not fh.is_all_in_sample(cutoff=self.cutoff):
```

Comment:
I feel like we can handle the in-sample vs out-of-sample predictions better on a higher-level in the class hierarchy but happy to leave it as it is for now.

Reply:
yes I agree, this higher-level function could just receive the in-sample and out-sample predictions and return a combined prediction.